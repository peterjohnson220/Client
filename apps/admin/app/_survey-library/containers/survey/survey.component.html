<pf-page-with-title-bar>
  <ng-container page-title>
    {{pageTitle}}
  </ng-container>

  <ng-container page-content>
    <pf-async-container [loading]="isLoadingData$ | async"
                        [loadingError]="(hasLoadingError$ | async) || deleteFailed "
                        [loadingErrorMessage]="loadingErrorMessage">
      <pf-input-debounce [delay]="400"
                         [(ngModel)]="tbxSearch"
                         (valueChanged)="handleSearchChanged()"
                         (clearClicked)="tbxSearch = ''">
      </pf-input-debounce>


      <table class="table table-hover">
        <thead>
          <tr>
            <th *ngIf="isUserGroupOne()">Survey</th>
            <th>ID</th>
            <th>Code</th>
            <th># Jobs</th>
            <th># Data Cuts</th>
            <th># Participants</th>
            <th># Published Matches</th>
            <th># Project Matches</th>
            <th>Map Companies</th>
            <th></th>
          </tr>
        </thead>
        <tbody>
          <tr *ngFor="let item of surveys">
            <td *ngIf="isUserGroupOne()">{{item.CompanyName}}</td>
            <td>{{item.SurveyId}}</td>
            <td>{{item.SurveyCode}}</td>
            <td>{{item.JobCount}}</td>
            <td>{{item.CutCount}}</td>
            <td>{{item.ParticipantCount}}</td>
            <td>{{item.PublishMatchCount}}</td>
            <td>{{item.ProjectMatchCount}}</td>
            <td (click)="mapCompaniesClick(item.SurveyId)"
                style="cursor: pointer"><u>{{item.CompanyCount}}</u></td>
            <td>
              <button type="button"
                      class="btn btn-info btn-sm"
                      (click)="copySurvey(item.SurveyId, item.CompanyName)"
                      [disabled]="item.CompanyHasSurveyUdfs"
                      [title]="(item.CompanyHasSurveyUdfs) ? 'Copy functionality is not available for this survey as it contains company specific custom fields' : ''">
                Copy Survey
              </button>
              &nbsp;
              <ng-container *ngIf="isUserGroupOne() && item.CompanyCount <= 1">
                <button type="button"
                        class="btn btn-danger btn-sm"
                        *ngIf="item.PublishMatchCount > 0 || item.ProjectMatchCount > 0"
                        (click)="deletePublishedMatches(item)">
                  Delete Published Matches
                </button>
                <button type="button"
                        class="btn btn-danger btn-sm"
                        *ngIf="item.PublishMatchCount === 0 && item.ProjectMatchCount === 0 && (item.JobCount > 0 || item.CutCount > 0)"
                        (click)="deleteSurvey(item.SurveyId, enumSurveyDelete.SURVEY_DATA)">
                  Delete Data Cuts
                </button>
                &nbsp;
                <button type="button"
                        class="btn btn-danger btn-sm"
                        *ngIf="item.PublishMatchCount === 0 && item.ProjectMatchCount === 0 && item.JobCount === 0 && item.CutCount === 0"
                        (click)="deleteSurvey(item.SurveyId, enumSurveyDelete.SURVEY)">
                  Delete Survey
                </button>
              </ng-container>
            </td>
          </tr>
        </tbody>
      </table>
      <button *ngIf="isUserGroupOne()"
              type="button"
              class="btn btn-sm"
              (click)="addSurvey()">Add Survey</button>
      &nbsp;
      <button type="button"
              class="btn btn-sm"
              (click)="backToSurveyYear()">Back to Survey Years</button>

    </pf-async-container>
  </ng-container>
</pf-page-with-title-bar>
<pf-add-survey-modal [surveyYearId]="surveyYearId">
</pf-add-survey-modal>
<pf-map-company-modal [surveyId]="selectedSurveyId">
</pf-map-company-modal>
<pf-copy-survey-modal [surveyId]="selectedSurveyId"
                      [SelectedSurveyCompany]="selectedCompany">
</pf-copy-survey-modal>
<pf-delete-confirmation-modal [publishedMatchCount]="selectedItemMatchCount"
                              [surveyId]="selectedSurveyId"></pf-delete-confirmation-modal>