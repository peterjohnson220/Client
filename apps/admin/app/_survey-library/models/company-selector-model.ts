export interface CompanySelectorItem {
  CompanyId: number;
  CompanyName: string;
}
