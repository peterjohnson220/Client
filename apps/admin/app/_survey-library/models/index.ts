export * from './survey-title-response-model';
export * from './survey-titles-filter.model';
