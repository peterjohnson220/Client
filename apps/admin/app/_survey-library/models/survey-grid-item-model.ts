export function generateMockSurveyGridItem() {
    return {
        AgingFactor: null,
        CompanyCount: 1,
        CompanyHasSurveyUdfs: true,
        CompanyId: 1,
        CompanyName: 'MockCompany',
        CutCount: 0,
        JobCount: 1,
        NotesCount: 3,
        ParticipantCount: 0,
        ProjectMatchCount: 0,
        PublishMatchCount: 0,
        SurveyCode: '5550Abc2019-10-07v1',
        SurveyId: 1,
        SurveyName: 'MockTest',
        SurveyYearId: 1,
        tbxSearch: undefined
    };
}
