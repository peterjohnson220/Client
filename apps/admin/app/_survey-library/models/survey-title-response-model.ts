export interface SurveyTitleResponseModel {
  SurveyTitleId: number;
  SurveyName: string;
}

export interface SurveyTitleCompanyModel {
  CompanyId: number;
  CompanyName: string;
  SurveyYearId: number;
  SurveyYear: number;
  CustomSurveyName: string;
}
