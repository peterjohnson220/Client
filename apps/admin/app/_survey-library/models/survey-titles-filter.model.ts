export interface SurveyTitlesFilter {
  SearchTerm: string;
  CompanyId: number;
}
