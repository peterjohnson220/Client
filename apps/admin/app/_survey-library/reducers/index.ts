import { createFeatureSelector, createSelector } from '@ngrx/store';

import * as fromRoot from 'libs/state/state';

import * as fromCustomSurveyTitleReducer from './custom-survey-title.reducer';
import * as fromCompanySelectorReducer from './company-selector.reducer';
import * as fromSurveyTitlesReducer from './survey-titles.reducer';
import * as fromSurveyReducer from './survey.reducer';

export interface SurveyLibraryState {
  customSurveyTitle: fromCustomSurveyTitleReducer.State;
  companySelector: fromCompanySelectorReducer.State;
  surveyTitles: fromSurveyTitlesReducer.State;
  surveyState: fromSurveyReducer.State;
}

export interface State extends fromRoot.State {
  survey_library: SurveyLibraryState;
}

export const reducers = {
  customSurveyTitle: fromCustomSurveyTitleReducer.reducer,
  companySelector: fromCompanySelectorReducer.reducer,
  surveyTitles: fromSurveyTitlesReducer.reducer,
  surveyState: fromSurveyReducer.reducer

};

export const selectSurveyLibraryState = createFeatureSelector<SurveyLibraryState>('survey_library');

export const selectCustomSurveyTitleState = createSelector(selectSurveyLibraryState,
  (state: SurveyLibraryState) => state.customSurveyTitle);
export const selectCompanySelectorState = createSelector(selectSurveyLibraryState,
  (state: SurveyLibraryState) => state.companySelector);
export const selectSurveyTitlesState = createSelector(selectSurveyLibraryState,
  (state: SurveyLibraryState) => state.surveyTitles);
export const selectSurveyState = createSelector(selectSurveyLibraryState,
  (state: SurveyLibraryState) => state.surveyState);

// Custom Survey Title area
export const getCustomSurveyTitleSaving = createSelector(selectCustomSurveyTitleState,
  fromCustomSurveyTitleReducer.getSavingCustomTitleSaving);

export const getCustomSurveyTitleSavingSuccess = createSelector(selectCustomSurveyTitleState,
  fromCustomSurveyTitleReducer.getSavingCustomTitleSavingSuccess);

export const getCustomSurveyTitleSavingError = createSelector(selectCustomSurveyTitleState,
  fromCustomSurveyTitleReducer.getSavingCustomTitleSavingError);

export const getSavedCustomTitleInfo = createSelector(selectCustomSurveyTitleState,
  fromCustomSurveyTitleReducer.getSavedInfo);

// Company selector area
export const getGettingCompanies = createSelector(selectCompanySelectorState,
  fromCompanySelectorReducer.getGettingCompanies);

export const getGettingCompaniesError = createSelector(selectCompanySelectorState,
  fromCompanySelectorReducer.getGettingCompaniesError);

export const getCompanies = createSelector(selectCompanySelectorState,
  fromCompanySelectorReducer.getCompanies);

// Survey titles area
export const getSavingSurveyTitles = createSelector(selectSurveyTitlesState,
  fromSurveyTitlesReducer.getSavingSurveyTitle);

export const getSavingSurveyTitlesSuccess = createSelector(selectSurveyTitlesState,
  fromSurveyTitlesReducer.getSavingSurveyTitleSuccess);

export const getSavingSurveyTitlesError = createSelector(selectSurveyTitlesState,
  fromSurveyTitlesReducer.getSavingSurveyTitleError);

export const getLoadingSurveyTitles = createSelector(selectSurveyTitlesState,
  fromSurveyTitlesReducer.getLoadingSurveyTitles);

export const getLoadingSurveyTitlesSuccess = createSelector(selectSurveyTitlesState,
  fromSurveyTitlesReducer.getLoadingSurveyTitlesSuccess);

export const getLoadingSurveyTitlesError = createSelector(selectSurveyTitlesState,
  fromSurveyTitlesReducer.getLoadingSurveyTitlesError);

export const getSurveyTitles = createSelector(selectSurveyTitlesState,
  fromSurveyTitlesReducer.getTitles);

export const getWarning = createSelector(selectSurveyTitlesState,
  fromSurveyTitlesReducer.getWarning);

// Survey Page
export const getSurveyData = createSelector(selectSurveyState, fromSurveyReducer.getSurveyData);
export const isLoadingSurveyData = createSelector(selectSurveyState, fromSurveyReducer.isLoadingSurveyData);
export const isMapCompaniesModalOpen = createSelector(selectSurveyState, fromSurveyReducer.isMapCompaniesModalOpen);
export const isCopySurveyModalOpen = createSelector(selectSurveyState, fromSurveyReducer.isCopySurveyModalOpen);
export const isDeleteConfirmationModalOpen = createSelector(selectSurveyState, fromSurveyReducer.isDeleteConfirmationModalOpen);
export const isLoadingMapModalData = createSelector(selectSurveyState, fromSurveyReducer.isLoadingMapModalData);
export const getMapModalDataFailed = createSelector(selectSurveyState, fromSurveyReducer.getMapModalDataFailed);
export const mapCompaniesModalData = createSelector(selectSurveyState, fromSurveyReducer.mapCompaniesModalData);
export const isAddSurveyModalOpen = createSelector(selectSurveyState, fromSurveyReducer.isAddSurveyModalOpen);
export const shouldRefreshGrid = createSelector(selectSurveyState, fromSurveyReducer.shouldRefreshGrid);
export const surveyLoadFailed = createSelector(selectSurveyState, fromSurveyReducer.surveyLoadFailed);

