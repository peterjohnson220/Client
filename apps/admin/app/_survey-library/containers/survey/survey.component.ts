import { Component, OnInit } from '@angular/core';

import { ActivatedRoute } from '@angular/router';

import { Store } from '@ngrx/store';
import { Observable } from 'rxjs';

import { SurveyLibraryApiService } from 'libs/data/payfactors-api/survey-library';
import { UserContext } from 'libs/models';
import * as fromRootState from 'libs/state/state';

import * as fromSurveyActions from '../../actions/survey-actions';
import * as fromSurveyState from '../../reducers';
import { EnumSurveyDelete } from '../../constants/survey-delete-enum';

@Component({
  selector: 'pf-survey',
  templateUrl: './survey.component.html',
  styleUrls: ['./survey.component.scss']
})
export class SurveyComponent implements OnInit {

  public surveyYearId: number;
  public pageTitle: string;
  public tbxSearch: string;
  public surveys: any;
  public surveyTitleId: number;
  public userContext$: Observable<UserContext>;
  private shouldRefreshGrid$: Observable<boolean>;
  private surveyData$: Observable<any>;
  public isLoadingData$: Observable<boolean>;
  public hasLoadingError$: Observable<boolean>;
  public systemUserGroupsId = -1;
  public selectedSurveyId: number;
  public selectedCompany: string;
  public enumSurveyDelete = EnumSurveyDelete;
  public deleteFailed = false;
  public selectedItemMatchCount = 0;

  public loadingErrorMessage = 'Failed to get survey data';

  constructor(
    private activeRoute: ActivatedRoute,
    private surveyApi: SurveyLibraryApiService,
    private rootStore: Store<fromRootState.State>,
    private store: Store<fromSurveyState.State>
  ) {
    this.surveyYearId = this.activeRoute.snapshot.params.id;
    this.userContext$ = this.rootStore.select(fromRootState.getUserContext);
    this.shouldRefreshGrid$ = this.store.select(fromSurveyState.shouldRefreshGrid);
    this.surveyData$ = this.store.select(fromSurveyState.getSurveyData);
    this.isLoadingData$ = this.store.select(fromSurveyState.isLoadingSurveyData);
    this.hasLoadingError$ = this.store.select(fromSurveyState.surveyLoadFailed);
  }

  ngOnInit() {
    this.surveyApi.getFirstSurveyByYearId(this.surveyYearId).subscribe(f => {
      const date = new Date(f.EffectiveDate);
      this.surveyTitleId = f.SurveyTitleId;
      this.pageTitle = f.SurveyPublisher + ' ' + f.SurveyName + ' ' +
        (date.getMonth() + 1) + '/' + date.getDate() + '/' + date.getFullYear();
    }
    );

    this.userContext$.subscribe(userContext => {
      this.systemUserGroupsId = userContext.CompanySystemUserGroupsId;
    });

    this.surveyData$.subscribe(f => {
      this.surveys = f;
    });

    this.shouldRefreshGrid$.subscribe(f => {
      if (f) {
        this.getSurveys();
      }
    });

    this.getSurveys();

  }

  isUserGroupOne(): boolean {
    // in case our usercontext isn't loaded yet don't give extra permissions
    if (this.systemUserGroupsId === -1) { return false; }
    return this.systemUserGroupsId === 1;
  }

  getSurveys() {
    this.store.dispatch(new fromSurveyActions.GetSurveys(this.surveyYearId, this.tbxSearch));
  }

  backToSurveyYear() {
    window.location.href = '/marketdata/admin/surveyyears.asp?surveytitle_id=' + this.surveyTitleId;
  }

  handleSearchChanged() {
    this.getSurveys();
  }

  addSurvey() {
    this.store.dispatch(new fromSurveyActions.SetAddSurveyModalOpen(true));
  }


  deleteSurvey(surveyId: number, step: EnumSurveyDelete) {

    this.selectedSurveyId = surveyId;

    let msg = '';
    switch (step) {
      case EnumSurveyDelete.PUBLISHED_MATCHES:
        this.store.dispatch(new fromSurveyActions.SetDeleteConfirmationModalOpen(true));
        return;
      case EnumSurveyDelete.SURVEY_DATA:
        msg = 'Are you sure you want to delete the associated jobs and cuts?';
        break;
      case EnumSurveyDelete.SURVEY:
        msg = 'Are you sure you want to delete this survey?';
        break;
      default:
        throw new Error('EnumSurveyDelete case not supported');
    }

    if (confirm(msg)) {
      this.surveyApi.deleteSurveyAndChildren(surveyId, step).subscribe(f =>
        this.getSurveys(), (err) => {
          this.deleteFailed = true;
          this.loadingErrorMessage = 'Failed to delete the survey or it\'s related records.';
        }
      );
    }
  }

  deletePublishedMatches(item: any) {
    this.selectedItemMatchCount = item.PublishMatchCount;
    this.selectedSurveyId = item.SurveyId;
    this.deleteSurvey(item.SurveyId, EnumSurveyDelete.PUBLISHED_MATCHES);
  }

  copySurvey(surveyId: number, company: string) {
    this.selectedSurveyId = surveyId;
    this.selectedCompany = company;
    this.store.dispatch(new fromSurveyActions.SetCopySurveyModalOpen(true));
  }

  public mapCompaniesClick(surveyId: number) {
    this.selectedSurveyId = surveyId;
    this.store.dispatch(new fromSurveyActions.GetMapCompaniesModalData(surveyId, ''));
    this.store.dispatch(new fromSurveyActions.SetMapCompaniesModalOpen(true));
  }
}
