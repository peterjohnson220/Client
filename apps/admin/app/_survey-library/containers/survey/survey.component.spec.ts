import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { ActivatedRoute } from '@angular/router';
import { NO_ERRORS_SCHEMA } from '@angular/core';

import { Store, StoreModule, combineReducers } from '@ngrx/store';
import { of } from 'rxjs';

import * as fromRootState from 'libs/state/state';
import { SurveyLibraryApiService } from 'libs/data/payfactors-api/survey-library';
import { generateMockUserContext } from 'libs/models';

import { SurveyComponent } from './survey.component';
import * as fromSurveyLibraryReducer from '../../reducers';

describe('Add Survey Titles modal', () => {
  let component: SurveyComponent;
  let fixture: ComponentFixture<SurveyComponent>;
  let store: Store<fromSurveyLibraryReducer.State>;
  let activatedRoute: ActivatedRoute;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        StoreModule.forRoot({
          ...fromRootState.reducers,
          survey_library: combineReducers(fromSurveyLibraryReducer.reducers)
        })
      ],
      declarations: [SurveyComponent],
      providers: [
        {
          provide: ActivatedRoute,
          useValue: { snapshot: { params: { id : 1 } } }
        },
        {
          provide: SurveyLibraryApiService,
          useValue: {
            // mock out getFirstSurveyByYearId to return a fake Observable
            getFirstSurveyByYearId: () => ({ subscribe: () => {} })
          }
        }
      ],
      schemas: [NO_ERRORS_SCHEMA]
    });
  }));

  beforeEach(() => {
    activatedRoute = TestBed.get(ActivatedRoute);
    store = TestBed.get(Store);
    fixture = TestBed.createComponent(SurveyComponent);
    component = fixture.componentInstance;
    component.userContext$ = of(generateMockUserContext());

    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should show an enabled copy button when the survey company does not have survey UDFs', () => {
    component.surveys = [{ CompanyHasSurveyUdfs: false }];
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should show a disabled copy button when the survey company has survey UDFs', () => {
    component.surveys = [{ CompanyHasSurveyUdfs: true }];
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});

