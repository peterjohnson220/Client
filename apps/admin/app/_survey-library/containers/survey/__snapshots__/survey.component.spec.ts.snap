// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Survey Titles modal should show a disabled copy button when the survey company has survey UDFs 1`] = `
<pf-survey
  __ngContext__={[Function LRootView]}
  activeRoute={[Function Object]}
  deleteFailed="false"
  enumSurveyDelete={[Function Object]}
  hasLoadingError$={[Function Store]}
  isLoadingData$={[Function Store]}
  loadingErrorMessage={[Function String]}
  rootStore={[Function Store]}
  selectedItemMatchCount="0"
  shouldRefreshGrid$={[Function Store]}
  store={[Function Store]}
  surveyApi={[Function Object]}
  surveyData$={[Function Store]}
  surveyNotesModal="undefined"
  surveyYearId={[Function Number]}
  surveys={[Function Array]}
  systemUserGroupsId={[Function Number]}
  userContext$={[Function Observable]}
>
  <pf-page-with-title-bar>
    <pf-async-container>
      <pf-input-debounce />
      <table
        class="table table-hover"
      >
        <thead>
          <tr>
            <th>
              Survey
            </th>
            <th>
              ID
            </th>
            <th>
              Code
            </th>
            <th>
              # Jobs
            </th>
            <th>
              # Data Cuts
            </th>
            <th>
              # Participants
            </th>
            <th>
              # Published Matches
            </th>
            <th>
              # Project Matches
            </th>
            <th>
              Map Companies
            </th>
            <th>
              Notes
            </th>
            <th />
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
            </td>
            <td>
            </td>
            <td>
            </td>
            <td>
            </td>
            <td>
            </td>
            <td>
            </td>
            <td>
            </td>
            <td>
            </td>
            <td>
              <u>
              </u>
            </td>
            <td>
              <u>
              </u>
            </td>
            <td>
              <button
                class="btn btn-info btn-sm"
                disabled=""
                title="Copy functionality is not available for this survey as it contains company specific custom fields"
                type="button"
              >
                 Copy Survey 
              </button>
            </td>
          </tr>
        </tbody>
      </table>
      <button
        class="btn btn-sm"
        type="button"
      >
        Add Survey
      </button>
      <button
        class="btn btn-sm"
        type="button"
      >
        Back to Survey Years
      </button>
    </pf-async-container>
  </pf-page-with-title-bar><pf-add-survey-modal /><pf-map-company-modal /><pf-copy-survey-modal /><pf-delete-confirmation-modal /><pf-survey-notes-modal />
</pf-survey>
`;

exports[`Add Survey Titles modal should show an enabled copy button when the survey company does not have survey UDFs 1`] = `
<pf-survey
  __ngContext__={[Function LRootView]}
  activeRoute={[Function Object]}
  deleteFailed="false"
  enumSurveyDelete={[Function Object]}
  hasLoadingError$={[Function Store]}
  isLoadingData$={[Function Store]}
  loadingErrorMessage={[Function String]}
  rootStore={[Function Store]}
  selectedItemMatchCount="0"
  shouldRefreshGrid$={[Function Store]}
  store={[Function Store]}
  surveyApi={[Function Object]}
  surveyData$={[Function Store]}
  surveyNotesModal="undefined"
  surveyYearId={[Function Number]}
  surveys={[Function Array]}
  systemUserGroupsId={[Function Number]}
  userContext$={[Function Observable]}
>
  <pf-page-with-title-bar>
    <pf-async-container>
      <pf-input-debounce />
      <table
        class="table table-hover"
      >
        <thead>
          <tr>
            <th>
              Survey
            </th>
            <th>
              ID
            </th>
            <th>
              Code
            </th>
            <th>
              # Jobs
            </th>
            <th>
              # Data Cuts
            </th>
            <th>
              # Participants
            </th>
            <th>
              # Published Matches
            </th>
            <th>
              # Project Matches
            </th>
            <th>
              Map Companies
            </th>
            <th>
              Notes
            </th>
            <th />
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
            </td>
            <td>
            </td>
            <td>
            </td>
            <td>
            </td>
            <td>
            </td>
            <td>
            </td>
            <td>
            </td>
            <td>
            </td>
            <td>
              <u>
              </u>
            </td>
            <td>
              <u>
              </u>
            </td>
            <td>
              <button
                class="btn btn-info btn-sm"
                title=""
                type="button"
              >
                 Copy Survey 
              </button>
            </td>
          </tr>
        </tbody>
      </table>
      <button
        class="btn btn-sm"
        type="button"
      >
        Add Survey
      </button>
      <button
        class="btn btn-sm"
        type="button"
      >
        Back to Survey Years
      </button>
    </pf-async-container>
  </pf-page-with-title-bar><pf-add-survey-modal /><pf-map-company-modal /><pf-copy-survey-modal /><pf-delete-confirmation-modal /><pf-survey-notes-modal />
</pf-survey>
`;
